### PR TITLE
BBA/Dima/1/feature/Created multiswitch component with tests

### PR DESCRIPTION
**Pull request by DmitriyBodlev**

### CODE REVIEW

File: src/components/multiswitch/Multiswitch.test.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  },
+  {
+    width: 50,
+    name: 'Bookers',
+    value: 'bookers',
+  },
+];
+
+it('Multiswitch should render and has two child components', () => {
+  const { container, getByTestId } = render(<Multiswitch selectedOptionIndex={0} options={multiswitchOptions} />);
```
Comment (DmitriyBodlev) [inline src/components/multiswitch/Multiswitch.test.js]: that it is on one line made not me, but beautifier when i committed changes.

File: src/components/multiswitch/Multiswitch.test.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  expect(container.childNodes.length).toEqual(1);
+  const childNodes = container.firstChild.childNodes;
+  expect(childNodes.length).toEqual(2);
+  expect(getByTestId(C.TEST_ID_MULTISWITCH)).toBeInTheDocument();
+});
+
+it('Multiswitch should change values after click on radio', () => {
+  const { container } = render(<Multiswitch selectedOptionIndex={0} options={multiswitchOptions} />);
```
Comment (DmitriyBodlev) [inline src/components/multiswitch/Multiswitch.test.js]: that it is on one line made not me, but beautifier when i committed changes.

File: src/components/multiswitch/Multiswitch.stories.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    name: 'Bookers',
+    value: 'drivers',
+  },
+];
+
+storiesOf('Multiswitch', module)
+  .addDecorator(story => <div style={{ padding: '3rem' }}>{story()}</div>)
+  .addDecorator(withInfo)
+  .addDecorator(withKnobs)
+  .addWithJSX('Multiswitch -> default', () => <Multiswitch selectedOptionIndex={0} options={multiswitchOptions} />);
```
Comment (DmitriyBodlev) [inline src/components/multiswitch/Multiswitch.stories.js]: that it is on one line made not me, but beautifier when i committed changes.

File: src/components/multiswitch/index.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import * as R from 'ramda';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { top, left, color, width, height, display, position, background, borderRadius } from 'styled-system';
```
Comment (DmitriyBodlev) [inline src/components/multiswitch/index.js]: that it is on one line made not me, but beautifier when i committed changes.

File: storybook-static/main.2d9ac116d17c61808bda.bundle.js [hunk 5]
```diff
@@ -0,0 +1 @@
+(window.webpackJsonp=window.webpackJsonp||[]).push([[0],[function(n,o,p){n.exports=p(289)}],[[0,1,2]]]);
```
Comment (DmitriyBodlev) [inline storybook-static/main.2d9ac116d17c61808bda.bundle.js:1]: what to do with this changes?

File: src/components/multiswitch/Multiswitch.test.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  const childNodes = container.firstChild.childNodes;
+  expect(childNodes[0].firstChild.checked).toEqual(true);
+  expect(childNodes[1].firstChild.checked).toEqual(false);
+  fireEvent.click(childNodes[1].firstChild);
+  expect(childNodes[0].firstChild.checked).toEqual(false);
+  expect(childNodes[1].firstChild.checked).toEqual(true);
+});
+
+it('Multiswitch should nas no children', () => {
```
Comment (VitaliyAndrushko) [inline src/components/multiswitch/Multiswitch.test.js]: should has

File: src/components/multiswitch/index.js [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+export const RadioBackground = styled.div`
+  ${top}
+  ${left}
+  ${width}
+  ${height}
+  ${position}
+  ${background}
+  ${borderRadius}
+  transition: 0.5s all, 0.5s border-radius .2s;
+  transform: translate(${({ forTransform }) => `${forTransform / 2}px`}, 6px);
```
Comment (VitaliyAndrushko) [inline src/components/multiswitch/index.js]: use R.divide

File: src/components/multiswitch/index.js [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+
+export const Radio = styled.input`
+  ${display}
+  &:checked ~ ${Label} {
+    ${color}
+  }
+  &:checked ~ ${RadioBackground} {
+    opacity: 1;
+    height: 12px;
+    width: 100%;
```
Comment (VitaliyAndrushko) [inline src/components/multiswitch/index.js:57]: cascade

File: src/components/multiswitch/index.js [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+const optionPropType = PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number]));
+
+RadioOption.propTypes = {
+  index: PropTypes.number,
+  radioOption: optionPropType,
+  selectedOptionIndex: PropTypes.number,
+};
+
+const MultiswitchComponent = props => (
+  <MultiswitchWrapper
```
Comment (VitaliyAndrushko) [inline src/components/multiswitch/index.js]: is it possible to have all props configurable?

File: storybook-static/main.2d9ac116d17c61808bda.bundle.js (hunk 5 above)
Comment (VitaliyAndrushko) [reply storybook-static/main.2d9ac116d17c61808bda.bundle.js:1]: checkout

File: src/components/multiswitch/index.js [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { top, left, color, width, height, display, position, background, borderRadius } from 'styled-system';
+// constants
+import * as C from '../../constants';
+// ui
+import { Flex, Box } from '../../ui';
+// /////////////////////////////////////////////////////////////////////////////////////////////////
+
+export const MultiswitchWrapper = styled(Flex)`
```
Comment (VitaliyAndrushko) [inline src/components/multiswitch/index.js:16]: add comment -> TODO: remove after adding global styles

File: src/components/multiswitch/index.js [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+import * as R from 'ramda';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { top, left, color, width, height, display, position, background, borderRadius } from 'styled-system';
+// constants
+import * as C from '../../constants';
+// ui
+import { Flex, Box } from '../../ui';
+// /////////////////////////////////////////////////////////////////////////////////////////////////
+
```
Comment (VitaliyAndrushko) [inline src/components/multiswitch/index.js:13]: Add comment -> TODO: with theme

File: src/components/multiswitch/index.js [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+export const MultiswitchWrapper = styled(Flex)`
+  box-sizing: border-box;
+`;
+
+export const RadioWrapper = styled(Box)`
```
Comment (VitaliyAndrushko) [inline src/components/multiswitch/index.js:20]: maybe add cursor to Box

File: src/components/multiswitch/index.js [hunk 13]
```diff
@@ -0,0 +1,188 @@
...
+    height: PropTypes.string.isRequired,
+    fontSize: PropTypes.string.isRequired,
+    checkedBg: PropTypes.string.isRequired,
+    fontWeight: PropTypes.string.isRequired,
+    checkedColor: PropTypes.string.isRequired,
+    borderRadius: PropTypes.string.isRequired,
+  }),
+};
+
+const makeWrapperSettings = settings => ({
```
Comment (VitaliyAndrushko) [inline src/components/multiswitch/index.js:135]: get colors from Theme and check design https://app.zeplin.io/project/5c31dd63323a8aaf34949cbf/screen/5d443194cd45a9887eede083

File: src/ui/abstract/index.js [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+  lineHeight,
+  borderLeft,
+  borderColor,
+  borderRight,
+  borderBottom,
+  borderRadius,
+} from 'styled-system';
+// //////////////////////////////////////////////////////////////////////////////
+
+export const Abstract = tag => styled(tag || 'div')`
```
Comment (VitaliyAndrushko) [inline src/ui/abstract/index.js]: R.or

File: src/ui/index.js [hunk 15]
```diff
@@ -1,9 +1,29 @@
-export { Box } from './box';
+import { or } from 'ramda';
```
Comment (VitaliyAndrushko) [inline src/ui/index.js]: import * as R

File: src/components/multiswitch/Multiswitch.stories.js [hunk 16]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
```
Comment (VitaliyAndrushko) [inline src/components/multiswitch/Multiswitch.stories.js]: remove this stories -> we skip all stories (save only icons)

File: src/components/text-ellipsed/TextEllipsed.stories.js [hunk 17]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
```
Comment (VitaliyAndrushko) [inline src/components/text-ellipsed/TextEllipsed.stories.js]: remove this file

File: src/constants/test-ids.js [hunk 18]
```diff
@@ -0,0 +1,11 @@
+export const TEST_ID_HOME_PAGE = 'HomePage';
```
Comment (VitaliyAndrushko) [inline src/constants/test-ids.js]: remove this file

File: src/ui/UI.stories.js [hunk 19]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
```
Comment (VitaliyAndrushko) [inline src/ui/UI.stories.js]: remove this file

File: src/ui/box/Box.stories.js [hunk 20]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
```
Comment (VitaliyAndrushko) [inline src/ui/box/Box.stories.js]: remove this file

**Pull request merged**